buildpack automation: split into frozen and rolling buildpacks

risicle opened the pull request:
What
----
https://www.pivotaltracker.com/story/show/185256162

~The frozen buildpacks being initialized as a copy of the then-current set of `cflinuxfs3` buildpacks.~ The rolling buildpacks being initialized as a copy of the then-current set of `cflinuxfs3` buildpacks. No frozen buildpacks initially - once we start introducing `cflinuxfs4`, the plan is to move the `cflinuxfs3` ones to `buildpacks.frozen.yml`

`buildpacks.yml` is now no longer intended to be edited by hand, instead entries can be added to `buildpacks.rolling.yml` or `buildpacks.frozen.yml` depending on whether auto-updating is desired.

Here I have **included** the buildpack bump present in #3204, as I don't want to have to regenerate that to accommodate this scheme change - we have already fully tested that.

~This also includes the initial set of `cflinuxfs4` buildpacks, based on the existing `cflinuxfs3` buildpacks trivially bumped to `cflinuxfs4`.~ Nope not anymore.

This has been tested using cherry-picked commits from the [184654180-apps-under-cflinuxfs4](https://github.com/alphagov/paas-cf/compare/184654180-apps-under-cflinuxfs4) branch, and has successfully deployed to `dev02` with platform apps being left on `cflinuxfs3` ([here](https://deployer.dev02.dev.cloudpipeline.digital/teams/main/pipelines/create-cloudfoundry/jobs/post-deploy/builds/315)). A successful experimental deployment of paas-admin on `cflinuxfs4` was made too ([here](https://deployer.dev02.dev.cloudpipeline.digital/teams/main/pipelines/create-cloudfoundry/jobs/deploy-paas-admin/builds/1753)).
 
How to review
-------------

:eyes: ?

Probably also worth testing the auto-buildpack-bump workflow. The result of it running successfully can be seen @ #3207

~This should probably not be merged before other `cflinuxfs4`-related changes have been merged such as those in the aforementioned [184654180-apps-under-cflinuxfs4](https://github.com/alphagov/paas-cf/compare/184654180-apps-under-cflinuxfs4) branch. Alternatively we could yank out the actual buildpack changes included in this, put a copy of the current `buildpacks.yml` in `buildpacks.rolling.yml` and leave `buildpacks.frozen.yml` empty until we're ready to make use of the feature.~

**This should not be merged before #3204**

---

🚨⚠️ Please do not merge this pull request via the GitHub UI ⚠️🚨

whi-tw reviewed (approved):
LGTM - code runs locally, does what I would expect it to!

risicle commented:
Ugh I just realized that `yq` will attempt to include comments from its constituent yamls and emit ugly condensed output :vomiting_face:

risicle commented:
Rebased now that #3204 is merged, making the diff to `buildpacks.yml` effectively null (other than meaningless indentation changes, which I dropped) apart from the added comment.
